server/chat.ts

```diff
@@ -839,15 +840,32 @@ export class CommandContext extends MessageContext {
 	can(permission: RoomPermission, target: User | null, room: Room): boolean;
 	can(permission: GlobalPermission, target?: User | null): boolean;
 	can(permission: string, target: User | null = null, room: Room | null = null) {
-		if (Users.Auth.hasPermission(this.user, permission, target, room, this.cmd, true)) return true;
+		if (Users.Auth.hasPermission(this.user, permission, target, room, this.fullCmd, true)) return true;
 		if (Users.Auth.hasPermission(this.user, permission, target, room, this.fullCmd, false)) {
 			// If we need to use the true group's permission, reset the visual group
 			this.user.resetVisualGroup();
 			return true;
 		}
+		if (this.handler!.isPrivate) return this.secretCommand();
 		this.errorReply(`${this.cmdToken}${this.fullCmd} - Access denied.`);
 		return false;
 	}
+	privatelyCan(permission: RoomPermission, target: User | null, room: Room): boolean;
+	privatelyCan(permission: GlobalPermission, target?: User | null): boolean;
+	privatelyCan(permission: string, target: User | null = null, room: Room | null = null) {
+		if (!this.handler!.isPrivate) {
+			// @ts-ignore (can be several things, depending)
+			return this.can(permission, target, room);
+		}
```

Comment:
```suggestion
		if (!this.handler!.isPrivate) {
			// @ts-ignore (can be several things, depending)
			return this.can(permission, target, room);
		}
```

This shouldn't happen, because `privatelyCan` should set `isPrivate`.

Reply:
Oh, okay